### ISSUE TITLE
node-role.kubernetes.io is restricted

### ISSUE THREAD

**Issue by BartVB** [jeroenvermeulen/hetzner-talos-k8s-rancher#1]
https://github.com/jeroenvermeulen/hetzner-talos-k8s-rancher/blob/master/deploy/talos-patch-worker.yaml#L4

Tries to change a restricted label, this causes errors like:

```
user: warning: [2024-06-10T20:25:49.591336063Z]: [talos] controller failed {"component": "controller-runtime", "controller": "k8s.NodeApplyController", "error": "2 error(s) occurred:\n\tnodes \"worker1.clustername\" is forbidden: is not allowed to modify labels: node-role.kubernetes.io/worker\n\tclient rate limiter Wait returned an error: rate: Wait(n=1) would exceed context deadline"}
```

But besides that; awesome work on this repo! Extremely helpful and educational when setting up a Talos/k8s cluster!

**Comment by jeroenvermeulen** [comment]
Thanks for reporting the issue. I will take a look.

**Comment by jeroenvermeulen** [comment]
This does work:
```
$ kubectl label node worker2.eu2 node-role.kubernetes.io/worker=true
node/worker2.eu2 labeled
```
Still checking...

**Comment by BartVB** [comment]
That works because then the control plane will take care of this. 

The issue is that the labels are applied by the node itself and the node is not allowed to change these `node-role` labels. If they could they could add themself to the control plane...

First do:

`$ kubectl label node worker2.eu2 node-role.kubernetes.io/worker`

Then try adding:

`node-role.kubernetes.io/worker: "true"`

to `machine.nodeLabels` in the `machineconfig` with `talosctl`.

When I do that the 'controller failed' error immediately re-appears in the console logs for that worker.

**Comment by jeroenvermeulen** [comment]
Should be fixed by f3c25cc71f7de1f77b1dfd1a493a6d08fd944edf
